test/ dir is in `include`, non-spec TS files (e.g. test utils) break build with v0.13 rootDir: `./src`

### Current Behavior

Prior to v0.13 #504 , I could put a file into my test folder that contained utility code for tests and it would get compiled without a problem.

`test/TestUtils.tsx`

Now, when I `tsdx build`, I see this error:

`rpt2: options error TS6059: File '/mui-rff/test/TestUtils.tsx' is not under 'rootDir' '/mui-rff/src'. 'rootDir' is expected to contain all source files.`

### Expected behavior

I have a place to put utility code for tests.

### Suggested solution(s)

No idea, but you have a list of 3rd parties using this project and it would be nice if you would start to test future versions of tsdx against these projects to see if you break peoples stuff and can then use that to find out how to tell people how to fix their broken stuff.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| TSDX           | v0.13 |
| TypeScript       | 3.8.3 |
| npm/Yarn         |  6.13.7 / 1.22.4 | 
| Node             |  v13.11.0 |
| Operating System | OSX Catalina |

--- a/jest.config.js
+++ b/jest.config.js
@@ -4,5 +4,7 @@ module.exports = {
   testPathIgnorePatterns: [
     '/node_modules/', // default
     '<rootDir>/templates/', // don't run tests in the templates
+    '<rootDir>/test/.*/fixtures/', // don't run tests in fixtures
+    '<rootDir>/stage-.*/', // don't run tests in auto-generated (and auto-removed) test dirs
   ],
 };

--- a/src/deprecated.ts
+++ b/src/deprecated.ts
@@ -22,7 +22,9 @@ export async function moveTypes() {
       'rootDir to "./src".\n' +
       'TSDX has deprecated setting tsconfig.compilerOptions.rootDir to ' +
       '"./" as it caused buggy output for declarationMaps and occassionally ' +
-      'for type declarations themselves.'
+      'for type declarations themselves.\n' +
+      'You may also need to change your include to remove "test", which also ' +
+      'caused declarations to be unnecessarily created for test files.'
   );
 
   try {

--- a/templates/basic/tsconfig.json
+++ b/templates/basic/tsconfig.json
@@ -1,5 +1,5 @@
 {
-  "include": ["src", "types", "test"],
+  "include": ["src", "types"],
   "compilerOptions": {
     "module": "esnext",
     "lib": ["dom", "esnext"],

--- a/templates/react-with-storybook/tsconfig.json
+++ b/templates/react-with-storybook/tsconfig.json
@@ -1,5 +1,5 @@
 {
-  "include": ["src", "types", "test"],
+  "include": ["src", "types"],
   "compilerOptions": {
     "module": "esnext",
     "lib": ["dom", "esnext"],

--- a/templates/react/tsconfig.json
+++ b/templates/react/tsconfig.json
@@ -1,5 +1,5 @@
 {
-  "include": ["src", "types", "test"],
+  "include": ["src", "types"],
   "compilerOptions": {
     "module": "esnext",
     "lib": ["dom", "esnext"],

--- a/test/e2e/fixtures/build-default/test/some-test.test.ts
+++ b/test/e2e/fixtures/build-default/test/some-test.test.ts
@@ -1,0 +1,3 @@
+// this is to test that .test/.spec files in the test/ dir are excluded
+
+// and that rootDir: './src' doesn't error with test/ files

--- a/test/e2e/fixtures/build-default/test/testUtil.ts
+++ b/test/e2e/fixtures/build-default/test/testUtil.ts
@@ -1,0 +1,4 @@
+// this is to test that test helper files in the test/ dir are excluded
+// i.e. files in test/ that don't have a .spec/.test suffix
+
+// and that rootDir: './src' doesn't error with test/ files

--- a/test/e2e/fixtures/build-default/types/blar.d.ts
+++ b/test/e2e/fixtures/build-default/types/blar.d.ts
@@ -1,0 +1,3 @@
+// this is to test that rootDir: './src' doesn't error with types/ files
+
+// and that declaration files aren't re-output in dist/

--- a/test/e2e/tsdx-build-default.test.js
+++ b/test/e2e/tsdx-build-default.test.js
@@ -31,6 +31,15 @@ describe('tsdx build :: zero-config defaults', () => {
     expect(output.code).toBe(0);
   });
 
+  it("shouldn't compile files in test/ or types/", () => {
+    const output = execWithCache('node ../dist/index.js build');
+
+    expect(shell.test('-d', 'dist/test/')).toBeFalsy();
+    expect(shell.test('-d', 'dist/types/')).toBeFalsy();
+
+    expect(output.code).toBe(0);
+  });
+
   it('should create the library correctly', () => {
     const output = execWithCache('node ../dist/index.js build');
 

